### PR TITLE
refactor: Remove some unnecessary uses of sorted sets

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

### CODE REVIEW

File: model/src/main/kotlin/utils/DependencyGraphBuilder.kt [hunk 1]
```diff
@@ -553,8 +553,8 @@ private fun constructSortedScopeMappings(
 ): Map<String, List<RootDependencyIndex>> {
     val orderedMappings = mutableMapOf<String, List<RootDependencyIndex>>()
 
-    scopeMappings.keys.toSortedSet().forEach { scope ->
-        orderedMappings[scope] = scopeMappings.getValue(scope)
+    scopeMappings.entries.sortedBy { it.key }.forEach { (scope, rootDependencyIndices) ->
+        orderedMappings[scope] = rootDependencyIndices
```
Comment (sschuberth) [inline model/src/main/kotlin/utils/DependencyGraphBuilder.kt:557]: Hmm, I believe we need a different rationale here, as the code does not really look simpler to me, but longer and more complex. What's the real goal here, I guess to remove the use of `toSortedSet()`? Then we should say so.

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: model/src/main/kotlin/utils/DependencyGraphBuilder.kt (hunk 1 above)
Comment (fviernau) [reply model/src/main/kotlin/utils/DependencyGraphBuilder.kt:557]: My goal was indeed to simplify. I've just dropped this commit.

**Review by sschuberth — APPROVED** [review]

File: model/src/main/kotlin/utils/DependencyGraphBuilder.kt (hunk 1 above)
Comment (sschuberth) [reply model/src/main/kotlin/utils/DependencyGraphBuilder.kt]: While reviewing the code base as part of https://github.com/oss-review-toolkit/ort/pull/8705 I've come across this place again, and I've changed my mind a bit:

While the new code still is not really simpler IMO, it's better as it avoids an unneeded `getValue()` on `scopeMappings` (and the mention of `toSortedSet()` which we prefer to get rid of).

So, mind re-introducing the commit with a slightly adjust commit message?

Comment (fviernau) [reply model/src/main/kotlin/utils/DependencyGraphBuilder.kt]: Sure, here you go: https://github.com/oss-review-toolkit/ort/pull/8706.

**Pull request merged**